core/src/test/scala/org/apache/spark/scheduler/BarrierCoordinatorSuite.scala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+import java.util.concurrent.TimeoutException
+
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+import org.scalatest.concurrent.Eventually
+
+import org.apache.spark._
+import org.apache.spark.rpc.RpcTimeout
+
+class BarrierCoordinatorSuite extends SparkFunSuite with LocalSparkContext with Eventually {
+
+  /**
+   * Get the current ContextBarrierState from barrierCoordinator.states by ContextBarrierId.
+   */
+  private def getBarrierState(
+      stageId: Int,
+      stageAttemptId: Int,
+      barrierCoordinator: BarrierCoordinator) = {
+    val barrierId = ContextBarrierId(stageId, stageAttemptId)
+    barrierCoordinator.states.get(barrierId)
+  }
+
+  test("normal test for single task") {
+    sc = new SparkContext("local", "test")
+    val barrierCoordinator = new BarrierCoordinator(5, sc.listenerBus, sc.env.rpcEnv)
+    val rpcEndpointRef = sc.env.rpcEnv.setupEndpoint("barrierCoordinator", barrierCoordinator)
+    val stageId = 0
+    val stageAttemptNumber = 0
+    rpcEndpointRef.askSync[Unit](
```

Comment:
We are still relying on the RPC framework, can we get rid of this?

Reply:
Sorry for missing this, done in 8cd78a9.